src/SessionSetup/ConnectToStadiaWidgetTest.cpp

```diff
@@ -38,4 +108,122 @@ TEST(ConnectToStadiaWidget, IsSetActive) {
   EXPECT_TRUE(radio_button->isEnabled());
 }
 
+TEST(ConnectToStadiaWidget, CallStartAndFail) {
+  {
+    const std::filesystem::path mock_ggp_working =
+        orbit_base::GetExecutableDir() / "OrbitMockGgpWorking";
+    ConnectToStadiaWidget widget{QString::fromStdString(mock_ggp_working)};
+    ErrorMessageOr<void> start_result = widget.Start();
+    std::string expected_error{
+        "Internal error: Unable to start ConnectToStadiaWidget, ssh_connection_artifacts_ is not "
+        "set."};
+    EXPECT_THAT(start_result, HasError(expected_error));
+    EXPECT_FALSE(widget.isEnabled());
+    auto* radio_button = widget.findChild<QRadioButton*>("radioButton");
+    ASSERT_NE(radio_button, nullptr);
+    EXPECT_THAT(radio_button->toolTip().toStdString(), expected_error);
+  }
+
+  {
+    ConnectToStadiaWidget widget{"non/existing/path/to/ggp"};
+
+    auto ssh_context = orbit_ssh::Context::Create();
+    ASSERT_TRUE(ssh_context.has_value());
+    DeploymentConfiguration deploy_config{NoDeployment{}};
+    SshConnectionArtifacts ssh_artifacts{&ssh_context.value(), ServiceDeployManager::GrpcPort{0},
+                                         &deploy_config};
+
+    widget.SetSshConnectionArtifacts(&ssh_artifacts);
+    ErrorMessageOr<void> start_result = widget.Start();
+    std::string error_substring{"Unable to use ggp cli"};
+    EXPECT_THAT(start_result, HasError(error_substring));
+    EXPECT_FALSE(widget.isEnabled());
+
+    auto* radio_button = widget.findChild<QRadioButton*>("radioButton");
+    ASSERT_NE(radio_button, nullptr);
+    EXPECT_THAT(radio_button->toolTip().toStdString(), testing::HasSubstr(error_substring));
+  }
+}
+
+TEST_F(ConnectToStadiaWidgetTestFixture,
+       StartWithoutPriorConnectionAndLoadInstancesAndReloadInstances) {
+  ErrorMessageOr<void> start_result = widget_->Start();
+  EXPECT_THAT(start_result, HasValue());
+
+  // After start (and after processing state machine transitions), the instances are loaded
+  // automatically. This means the overlay is visible, the refresh button is disabled and the
+  // instance list is empty;
+  QApplication::processEvents();
+  EXPECT_TRUE(overlay_->isVisible());
+  EXPECT_FALSE(refresh_button_->isEnabled());
+  EXPECT_EQ(instances_table_view_->model()->rowCount(), 0);
+
+  WaitForMockGgp();
```

Comment:
I assume we could avoid this by mocking `orbit_ggp::Client`, right? Have you considered this? I assume it's just too much work for now, right?

Reply:
Also, did you run this 10,000 times or so, to be sure it's not flaky?